tpcc/main.go

```diff
@@ -180,14 +181,17 @@ func main() {
 		}
 	} else {
 		// 10 workers per warehouse.
-		workers = make([]*worker, *warehouses*10)
-		for i := 0; i < *warehouses; i++ {
-			for j := 0; j < 10; j++ {
-				idx := i*10 + j
-				workers[idx] = newWorker(idx, db, &wg)
-				go workers[idx].run(errCh, i)
+		go func() {
+			workers = make([]*worker, 0, *warehouses*10)
```

Comment:
Since this is happening in a goroutine, any access outside of the goroutine will be racey.